### ISSUE TITLE
[API] Set up user roles in the backend

### ISSUE THREAD

**Issue by bdougie** [open-sauced/app#110]
We need to identify user types or roles in the DB. At the moment all users are given same access. I am proposing the following roles get added to Supabase.

- Admin - Open Sauced employees that need access to update data
- Digital Ocean - This will ideally only get additional access to reporting (for now)
- Insider - 50 customers we will be getting beta feedback from
- User - This will be users not part of the insiders program and potentially onboarded from hot.opensauced.pizza 

Open to suggestions 

### Blocked by:
- #395
